### PR TITLE
Fixed InvalidCastException when using a sqlite database

### PR DESCRIPTION
**Pull request by icnocop**
Additional fixes for #42

### CODE REVIEW

**Comment by icnocop** [comment]
This is the result of how sqlite stores its data, and therefore retrieves its data.

Basically, the data types are dynamic and if the numeric data is larger than `Int32.MaxValue`, it will be retrieve with `GetInt64` for example.

https://www.sqlite.org/datatype3.html#affinity

File: src/EFCache/CachingReader.cs [hunk 1]
```diff
@@ -137,7 +137,16 @@ public override short GetInt16(int ordinal)
...
+            var value = GetValue(ordinal);
+            switch (value)
+            {
+                case int intValue:
+                    return intValue;
+                case long longValue:
+                    return Convert.ToInt32(longValue);
+                default:
+                    throw new NotImplementedException($"{value.GetType()}");
+            }
```
Comment (moozzyk) [inline src/EFCache/CachingReader.cs:149]: Any reason why we cannot do:

```suggestion
            return Convert.ToInt32(GetValue(ordinal));
```

File: test/EFCacheTests/CachingReaderTests.cs [hunk 2]
```diff
@@ -1206,6 +1219,7 @@ private static IEnumerable<object[]> GetRows(int count)
                 row[10] = new DateTime(2001, 11, 17);
                 row[11] = (byte)15;
                 row[12] = DBNull.Value;
+                row[13] = (long)0;
```
Comment (moozzyk) [inline test/EFCacheTests/CachingReaderTests.cs:1222]:
```suggestion
                row[13] = 0L;
```

**Comment by icnocop** [comment]
Thank you. I've updated the code based on your suggestions.

**Pull request merged**